### PR TITLE
Ali Gok_React_week3

### PR DESCRIPTION
**Pull request by aligoknl**

### CODE REVIEW

**Comment by crevulus** [comment]
Hi Ali! I haven't had the chance to go through the code yet, fully yet, but I see from your deployed link that you have done an excellent job again this week. Your React prep has certainly paid off! My recommendation for your project would be to experiment with styling more. Styling was not a requirement for this module and you have absolutely done the right thing by getting really good at the foundations of React before focusing on styling. For your portfolio projects, a well-styled site will look great to prospective employers, so do pay just a little more attention to creating a beautiful UI!

File: week3/project/ecommerce/src/fetches/useFetchData.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import { useEffect, useState } from "react";
```
Comment (crevulus) [inline week3/project/ecommerce/src/fetches/useFetchData.js:1]: `/fetches` is an unconventional name for a folder. Think about it: if you have hooks for other functionality, e.g. copying, are you going to make a folder called `/copies`? `/hooks` is a more conventional folder name. 🟢

File: week3/project/ecommerce/src/components/FavoriteContext.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import { createContext, useState } from "react";
```
Comment (crevulus) [inline week3/project/ecommerce/src/components/FavoriteContext.js:1]: `Context` usually doesn't come under the `/component` folder. Is it a component, but it's a *higher order component*, i.e. it wraps around other components and provides data or functionality. We'd usually put this in `/store`, `/data`, or even `/context`. 🟢

File: week3/project/ecommerce/src/components/FavoriteContext.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+export const FavoriteContext = createContext();
+
+export const FavoriteProvider = (props) => {
+  const [favorite, setFavorite] = useState([]);
+
+  const handleFavorite = (id) => {
```
Comment (crevulus) [inline week3/project/ecommerce/src/components/FavoriteContext.js:7]: Very nice!

File: week3/project/ecommerce/src/components/FavoriteContext.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import { createContext, useState } from "react";
+export const FavoriteContext = createContext();
+
+export const FavoriteProvider = (props) => {
+  const [favorite, setFavorite] = useState([]);
```
Comment (crevulus) [inline week3/project/ecommerce/src/components/FavoriteContext.js:5]: `favorite` is actually a list, so consider naming it `favorites` for more clarity. 🟢

File: week3/project/ecommerce/src/components/ProductsItem.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+function ProductsItem({ item }) {
+  const [favorite, handleFavorite] = useContext(FavoriteContext);
+  const favoriteItem = favorite.find((favoriteId) => favoriteId === item.id);
+
+  return (
+    <li className="products-item">
+      <div className="product">
+        <div className="heart-image" onClick={() => handleFavorite(item.id)}>
+          {favoriteItem ? <SolidHeart /> : <RegularHeart />}
```
Comment (crevulus) [inline week3/project/ecommerce/src/components/ProductsItem.js:15]: Nice!

**Review by crevulus — COMMENTED** [review]
Wow, once again Ali, no 🔴 notes. Very impressive! I think changing some of your structuring will help your colleagues understand your code better, so my comments are mostly focused on this.

File: week3/project/ecommerce/src/fetches/useFetchData.js (hunk 1 above)
Comment (aligoknl) [reply week3/project/ecommerce/src/fetches/useFetchData.js:1]: > `/fetches` is an unconventional name for a folder. Think about it: if you have hooks for other functionality, e.g. copying, are you going to make a folder called `/copies`? `/hooks` is a more conventional folder name. 🟢

I have changed folder's name as `/hooks`

File: week3/project/ecommerce/src/components/FavoriteContext.js (hunk 2 above)
Comment (aligoknl) [reply week3/project/ecommerce/src/components/FavoriteContext.js:1]: > `Context` usually doesn't come under the `/component` folder. Is it a component, but it's a _higher order component_, i.e. it wraps around other components and provides data or functionality. We'd usually put this in `/store`, `/data`, or even `/context`. 🟢

I have changed folder's name as `/context`

File: week3/project/ecommerce/src/components/FavoriteContext.js (hunk 4 above)
Comment (aligoknl) [reply week3/project/ecommerce/src/components/FavoriteContext.js:5]: > `favorite` is actually a list, so consider naming it `favorites` for more clarity. 🟢

I have changed as `favorites`

File: week3/project/ecommerce/src/components/FavoriteContext.js (hunk 3 above)
Comment (aligoknl) [reply week3/project/ecommerce/src/components/FavoriteContext.js:7]: > Very nice!

Thank you so much!

File: week3/project/ecommerce/src/components/ProductsItem.js (hunk 5 above)
Comment (aligoknl) [reply week3/project/ecommerce/src/components/ProductsItem.js:15]: > Nice!

Thank you so much!

File: week3/project/ecommerce/src/fetches/useMultipleFetchData.js [hunk 6]
```diff
@@ -0,0 +1,27 @@
+import { useEffect, useState } from "react";
```
[the remark this answers is not in the document]
Comment (aligoknl) [reply week3/project/ecommerce/src/fetches/useMultipleFetchData.js:1]: I have deleted `useMultipleFetchData` file and added the code into `useFetchData`

**Pull request closed without merging**